src/components/legendComponents/CategoryList.tsx

```diff
@@ -11,32 +11,50 @@ import {
   ListItemText
 } from '@mui/material'
 
-import React, { useEffect } from 'react'
+import React from 'react'
 
-import { APIManager } from '@/utils/APIManager'
 import { icons } from '@/store/Icons'
 
-const CategoryList = () => {
-  const {
-    selected,
-    categories,
-    setCategories,
-    handleAll,
-    handleNone,
-    handleChange
-  } = useCalendarContext()
+const CategoryList = ({
+  selectedNotSaved,
+  setSelectedNotSaved
+}: {
+  selectedNotSaved: Category[]
+  setSelectedNotSaved: React.Dispatch<React.SetStateAction<Category[]>>
+}) => {
+  const { categories, setSelected } = useCalendarContext()
 
-  useEffect(() => {
-    const getData = async () => {
-      const instance = await APIManager.getInstance()
-      const data = await instance.getCategory()
-      const cat: Category[] = data.result
-      setCategories(cat)
-    }
-    getData().catch((err) => {
-      console.log(err)
-    })
-  }, [])
+  const handleChange = (category: { target: { value: any } }) => {
+    const s: string = category.target.value
+    const list = [...selectedNotSaved]
+    const index = list
+      .map(function (e: Category) {
```

Comment:
instead of map then indexOf could just use findIndex. That way it doesn't have to go through the array twice.